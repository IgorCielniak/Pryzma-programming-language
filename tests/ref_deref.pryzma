struct Person {
    name = "Alice"
    age = 0
}

/greet{
    result = args[0]
    print "hello " + result.name, "\n"
    *result.age = 10
}

p = Person {}

@greet(&p)

print p.age,"\n"

r = p

r.name = "ff"

print p.name,"\n"

print r,"\n"


x = 8
p = &x
*p = 10

print x

x = 8
y = 12
p = &x
p = &y
*p = 14
print y

p = "gg"

print p, y
