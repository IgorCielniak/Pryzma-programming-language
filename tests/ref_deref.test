python Pryzma.py tests/ref_deref.pryzma
