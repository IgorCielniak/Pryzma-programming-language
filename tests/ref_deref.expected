hello Alice
10
Alice
{'name': 'ff', 'age': 10}
1014gg14